src/ui/common/src/utils/shared.ts

```diff
@@ -13,11 +13,18 @@ export type LoadingStatus = {
 };
 
 export enum ExecutionStatus {
+  Unknown = 'unknown',
   Succeeded = 'succeeded',
   Failed = 'failed',
   Pending = 'pending',
 }
 
+export enum FailureReason {
```

Comment:
Couple tiny nits here -- can we make the default the success type rather than a system error.

Is there a reason that we want to maintain this separately from the execution status? It seems a little confusing to me -- under execution status, would we rather have pending, succeeded, system error, user error, and (soon) warning? That seems more natural to me rather than splitting the context. Thoughts?

Reply:
I've been moving back and forth on this. I do find the current implementation a bit easier to handle, the main reason is that we are duplicating this status handling in a few places, including executor, go, sdk, and frontend.

The most important use of `ExecutionStatus` enum is to determine if the execution failed by checking `status == ExecutionStatus.Failed`. If we merge the status with failure reason, that means all 4 places need to implement some helper like `isFailed(status)` and check something like `status in [ExecutionStatus.UserFailure, ExecutionStatus.SystemFailure]` which is harder to maintain if we expand on failure types.

I think to address the confusion we could use 'NotAvailable' or 'Unknown', instead of 'NoFailure'. In this way it's more clear that this field is only useful in case of failure, and success / fail check should be done based on the status field.

Reply:
That's a fair point. In that case, let's rename this slightly to `FailureType`, and we can leave it as is. I think we can just call the default value "Success" because it's clear that Success is a non-existent type of failure.